common/enum_base.h

```diff
@@ -51,7 +51,8 @@ namespace Carbon::Internal {
 //   #include ".../my_kind.def"
 //   };
 //   ```
-template <typename DerivedT, typename EnumT>
+using EnumBaseNamesT = const llvm::StringLiteral[];
+template <typename DerivedT, typename EnumT, EnumBaseNamesT Names>
```

Comment:
Having the `using` first makes the big comment block seem to pertain to the `using` rather than the class template. And I don't think we really use it -- what about just inlining the type?

```suggestion
template <typename DerivedT, typename EnumT, const llvm::StringLiteral Names[]>
```

You could also make it a pointer if that seems more clear (I believe that is how this syntax works in C++ anyways):

```suggestion
template <typename DerivedT, typename EnumT, const llvm::StringLiteral* Names>
```

Reply:
Fixed (1st suggestion). I'd done this because I'd been getting a compile error, but I think I was just messing up my syntax and failed to realize.